### PR TITLE
protocol-auto模式下 dubbo stream filter适配 #1667

### PR DESCRIPTION
**Pull request by Thiswang**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: pkg/filter/stream/dubbo/dubbo.go [hunk 1]
```diff
@@ -46,8 +46,8 @@ func (d *dubboFilter) OnReceive(ctx context.Context, headers api.HeaderMap, buf
 
 	service, ok := headers.Get(dubbo.ServiceNameHeader)
 	if !ok {
-		log.DefaultLogger.Errorf("%s is empty, may be the protocol is not dubbo", dubbo.ServiceNameHeader)
-		return api.StreamFiltertermination
+		log.DefaultLogger.Warnf("%s is empty, may be the protocol is not dubbo", dubbo.ServiceNameHeader)
```
Comment (taoyuanyuan) [inline pkg/filter/stream/dubbo/dubbo.go:49]: 这样改不太妥，如果是dubbo协议的话，这个确实是必须的，不能continue。
感觉可以在最开始通过types.ContextSubProtocol来判断是否为dubbo，不为dubbo就continue

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Review by nejisama — APPROVED** [review]

**Comment by nejisama** [comment]
LGTM

File: pkg/filter/stream/dubbo/dubbo.go [hunk 2]
```diff
@@ -42,6 +42,11 @@ func (d *dubboFilter) OnDestroy() {}
 
 func (d *dubboFilter) OnReceive(ctx context.Context, headers api.HeaderMap, buf buffer.IoBuffer, trailers api.HeaderMap) api.StreamFilterStatus {
 
+	subProtocol := mosnctx.Get(ctx, types.ContextSubProtocol)
+	if subProtocol == nil || "dubbo" != subProtocol.(string) {
```
Comment (champly) [inline pkg/filter/stream/dubbo/dubbo.go:46]: "dubbo" replace with `pkg/protocol/xprotocol/dubbo/types.go` .`ProtocolName`

**Pull request merged**